src/GoogleFit/FitOnFhir.GoogleFit.Tests/GoogleFitTokensServiceTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using FitOnFhir.Common.Repositories;
+using FitOnFhir.Common.Tests.Mocks;
+using FitOnFhir.GoogleFit.Client.Responses;
+using FitOnFhir.GoogleFit.Services;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using Xunit;
+
+namespace FitOnFhir.GoogleFit.Tests
+{
+    public class GoogleFitTokensServiceTests
+    {
+        private const string _googleUserId = "me";
+        private const string _accessToken = "AccessToken";
+        private const string _refreshToken = "RefreshToken";
+        private readonly AuthTokensResponse _tokensResponse = new AuthTokensResponse() { AccessToken = _accessToken, RefreshToken = _refreshToken };
+        private readonly CancellationToken _cancellationToken = CancellationToken.None;
+
+        private readonly IUsersKeyVaultRepository _usersKeyvaultRepository;
+        private readonly IGoogleFitAuthService _googleFitAuthService;
+        private readonly MockLogger<GoogleFitTokensService> _tokensServiceLogger;
+        private readonly IGoogleFitTokensService _googleFitTokensService;
+
+        public GoogleFitTokensServiceTests()
+        {
+            // GoogleFitTokensService dependencies
+            _usersKeyvaultRepository = Substitute.For<IUsersKeyVaultRepository>();
+            _googleFitAuthService = Substitute.For<IGoogleFitAuthService>();
+            _tokensServiceLogger = Substitute.For<MockLogger<GoogleFitTokensService>>();
+
+            // create the service
+            _googleFitTokensService = new GoogleFitTokensService(_usersKeyvaultRepository, _googleFitAuthService, _tokensServiceLogger);
+        }
+
+        [Fact]
+        public async Task GivenValidGoogleFitId_WhenRefreshTokenIsCalled_GetByNameIsCalled()
+        {
+            SetupMockSuccessReturns();
+
+            await _googleFitTokensService.RefreshToken(_googleUserId, _cancellationToken);
+
+            _ = _usersKeyvaultRepository.Received(1).GetByName(
```

Comment:
btw: instead of doing _ = _users... You can just use await.

e.g.

```c#
await _googleFitAuthService.Received(1).RefreshTokensRequest(
    Arg.Is<string>(refresh => refresh == _refreshToken),
    Arg.Is<CancellationToken>(cancel => cancel == _cancellationToken));
```